libs/constrained-generators/src/Constrained/Base.hs

```diff
@@ -5367,13 +5457,17 @@ class Sized t where
 instance Ord a => Sized (Set.Set a) where
   sizeOf = toInteger . Set.size
   liftSizeSpec spec cant = typeSpec (SetSpec mempty TrueSpec (TypeSpec spec cant))
-  liftMemberSpec xs = typeSpec (SetSpec mempty TrueSpec (MemberSpec (nubOrd xs)))
+  liftMemberSpec xs = case NE.nonEmpty xs of
```

Comment:
There is some repeated code here, possibly we want to do this at the callsite instead?

Reply:
Actually, the code is only similar.
The ErrorSpec carry different messages
and the TypeSpecs are different. One is at type List the other at type Set,
Though I agree they look the same superficially, they are quite different.